android/src/main/java/com/theoplayer/ads/AdsModule.kt

```diff
@@ -133,4 +140,48 @@ class AdsModule(context: ReactApplicationContext) : ReactContextBaseJavaModule(c
       }
     }
   }
+
+  @ReactMethod
+  fun addFriendlyObstruction(tag: Int, obstruction: ReadableMap) {
+    val obsTag =
+      if (obstruction.hasKey(PROP_OMID_VIEW)) obstruction.getInt(PROP_OMID_VIEW) else null
+    val purpose = if (obstruction.hasKey(PROP_OMID_PURPOSE)) {
+      when (obstruction.getString(PROP_OMID_PURPOSE)) {
+        "videoControls" -> OmidFriendlyObstructionPurpose.VIDEO_CONTROLS
+        "closeAd" -> OmidFriendlyObstructionPurpose.CLOSE_AD
+        "notVisible" -> OmidFriendlyObstructionPurpose.NOT_VISIBLE
+        else -> OmidFriendlyObstructionPurpose.OTHER
+      }
+    } else null
+    val reason =
+      if (obstruction.hasKey(PROP_OMID_REASON)) obstruction.getString(PROP_OMID_REASON) else null
+
+    if (obsTag !== null) {
+      viewResolver.resolveViewByTag(obsTag) { obsView: View? ->
+        addFriendlyObstruction(tag, obsView, purpose, reason)
+      }
+    }
+  }
+
+  private fun addFriendlyObstruction(
+    tag: Int,
+    obsView: View?,
+    purpose: OmidFriendlyObstructionPurpose?,
+    reason: String?
+  ) {
+    if (obsView != null && purpose != null) {
```

Comment:
prob decrease level of nesting by if-invertion?
```suggestion
    if (obsView == null || purpose == null) {
      return
    }
```